doc/md/_a4_Getting-Started-ADCs.md

```diff
@@ -0,0 +1,173 @@
+
+    // Wait half a second (500 milliseconds)
+    System::Delay(500);
+  }
+}
+```
+
+To find out more about the serial printing, and what software to use, see our [tutorial on working with Serial Printing](_a2_Getting-Started-Serial-Printing.md).
+
+With the above program, you should see a number printing that moves from 0 to 65536 as you rotate the pot. Now, pots, and electronics aren't perfect.
+So there's a chance that this may not reach those extremes, or it might reach them slightly before the edges of the potentiometer's range.
+This will largely depend on the potentiometer you're using.
+
+Now that's great and all, but it would be nicer to work with a number that is scaled between 0 and 1 instead (since that's more useful for doing other things).
+
+We can actually get the value in that range instead by changing the line where we were doing:
+
+```cpp
+int value = hw.adc.Get(0); // value will be 0-65536
+```
+
+to this Instead:
+
+```cpp
+float value = hw.adc.GetFloat(0); // value will be 0.0 to 1.0
+```
+
+floating point numbers have a few pecularities because of the hardware and memory requirements of an embedded project, but you can keep the serial printing by adding the following to the project Makefile:
+
+```make
+LDFLAGS += -u _printf_float
+```
+
+This isn't included by default because it increases the flash size by several kilobytes, and there are other ways to print floating point numbers (more info on this in the [serial printing tutorial](tutorial)).
+
+## Multiple Inputs
+
+Most things get better when you add more of whatever it is. The same goes for controls!
+
+Now we've set up a single pot. So let's look at what it would look like to hook up two (or more).
+
+Nearly everything will stay the same except instead of using a single `AdcChannelConfig`, we'll want to create one for each control we want to use. C and C++ have a mechanism for dealing with this known as an array. If you're not familiar, it is basically just an ordered list.
+
+So for setting up two pots we'll change the initialization code to look a little more like:
+
+```cpp
+// Create an array of two AdcChannelConfig objects
+AdcChannelConfig my_adc_config[2];
+// Initialize the first one connected to A0
+my_adc_config[0].InitSingle(A0);
+// Initialize the second one connected to A4
+my_adc_config[1].InitSingle(A4);
+```
+
+And finally when we go to initialize the ADC itself, we can now omit the `&` that we had to use before because it's an array of objects, and we want to make sure to update the total count that we passed in before:
+
+```cpp
+hw.adc.Init(my_adc_config, 2);
+```
+
+Now you can read from either input. If you want to read the pot connected to A0 you would use `hw.adc.Get(0)`, or `hw.adc.GetFloat(0)`, and if you want to read from the pot connected to A4, you would use `hw.adc.Get(1)`, or `hw.adc.GetFloat(1)`.
+
+You'll notice that you access them by the order in which you initialized them, not by the number associated with the Pin.
+
```

Comment:
## Order to chaos - naming ADC channels

Once you have a lot of ADC channels, it can be difficult to remember which channel corresponds to which control. To solve this, you can use `enum`s to assign names to channels like this:
```cpp
enum AdcChannel {
   pitchKnob = 0,
   pitchCv,
   gainKnob,
   gainCv,
   NUM_ADC_CHANNELS
}
```

Here, the first value is assigned to `0`and the others increment automatically. Interestingly, this logic can be used to automatically get the total number of channels you use, by adding one last value `NUM_ADC_CHANNELS`. Of course you could also specify the individual values by hand, just add `myKnob = 3,`.

Now you can make your code much easier to read - and maintain, should you decide to change the hardwre connections later:
```cpp
AdcChannelConfig my_adc_config[NUM_ADC_CHANNELS];
// ...
my_adc_config[pitchCv].InitSingle(A0);
// ...
hw.adc.Init(my_adc_config, NUM_ADC_CHANNELS);
// ...
const pitchCvValue = hw.adc.GetFloat(pitchCv);
```

Reply:
Another great suggestion. Added as well!